pycqed/scripts/Spectroscopy.py

```diff
@@ -0,0 +1,281 @@
+"""
+Spectroscopy class
+
+This file contains the Spectroscopy class that forms the basis analysis of all the spectroscopy measurement analyses.
+"""
+import pycqed.analysis_v2.base_analysis as ba
+import numpy as np
+import pandas as pd
+from pycqed.analysis import measurement_analysis as MA
+from pycqed.analysis import analysis_toolbox as a_tools
+from pycqed.analysis.tools import data_manipulation as dm_tools
+from pycqed.analysis import fitting_models as fit_mods
+import lmfit
+
+class Spectroscopy(ba.BaseDataAnalysis):
```

Comment:
A docstring describing that (if) this is an abstract base class and what it is used for/how it should be used would be quite helpful here.